openhtf/plugs/__init__.py

```diff
@@ -239,13 +241,14 @@ def discover(cls, host, port, timeout_s=xmlrpcutil.DEFAULT_PROXY_TIMEOUT_S):
     This is so that the yielded tuples may be easily passed to sockjs a la:
         sockjs.tornado.SockJSRouter(*result)
     """
-    proxy = xmlrpcutil.TimeoutProxyMixin((host, port))
+    proxy = xmlrpcutil.TimeoutProxyServer('http://%s:%s' % (host, port),
+                                          timeout_s=timeout_s, allow_none=True)
     seen = set()
     for method in proxy.system.listMethods():
       if not method.startswith('plugs.'):
         continue
       try:
-        _, plug_name, _ = method.split('.')
+        plug_name = method.rsplit('.', 1)[0].split('.', 1)[1]
```

Comment:
oh god, can you add a comment with an example of what `method` looks like to make this a bit more human parseable later?  i realize this was my derpy doing in the first place, sorry about that x.x